libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -178,7 +179,6 @@ - (void)dealloc {
 - (void)authenticate {
     NSAssert(nil != self.credentials, @"credentials cannot be nil");
     NSAssert(self.credentials.clientId.length > 0, @"credentials.clientId cannot be nil or empty");
-    NSAssert(self.credentials.identifier.length > 0, @"credentials.identifier cannot be nil or empty");
```

Comment:
I don't think this should be removed.  Credentials should have a valid identifier before attempting authentication.